File: LinearSearch.py
# Algorithm for Linear Search in Python

# Steps ->
# 1. Start from the leftmost element of arr[] and one by one compare it with each element of arr[].
# 2. If element matches with an element, return the index.
# 3. If element does not match with any of elements, return -1.

array = [8, 14, 15, 17, 20, 2, 4, 6, 9, 18]

def linearSearch(array, key):
    for i in range(len(array)):
        if array[i] == key:
            return i
    return -1

x = linearSearch(array, 4)
print(x)

File: linearsearch.cpp
#include <iostream>
using namespace std;

int search(int arr[], int a, int b)
{
	int i;
	for (i = 0; i < a; i++)
		if (arr[i] == b)
			return i;
	return -1;
}
int main(void)
{
	int arr[] = { 2, 3, 4, 10, 40 };
	int b = 10;
	int a = sizeof(arr) / sizeof(arr[0]);
	int result = search(arr, a, b);
	(result == -1)? cout << "Element not found": cout << "Element found " << result;
	return 0;
}
